pkg/resource_builders/pod/externalsecrets.go

```diff
@@ -0,0 +1,93 @@
+package pod
+
+import (
+	"fmt"
+	"reflect"
+
+	externalsecretsv1alpha1 "github.com/3scale/saas-operator/pkg/apis/externalsecrets/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GenerateExternalSecretFn generates a ExternalSecret
+func GenerateExternalSecretFn(name, namespace, secretStoreName, secretStoreKind string, refreshInterval metav1.Duration, labels map[string]string,
+	opts interface{}) func() *externalsecretsv1alpha1.ExternalSecret {
+
+	return func() *externalsecretsv1alpha1.ExternalSecret {
+		return &externalsecretsv1alpha1.ExternalSecret{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       externalsecretsv1alpha1.ExtSecretKind,
+				APIVersion: externalsecretsv1alpha1.ExtSecretGroupVersionKind.GroupVersion().String(),
+			},
```

Comment:
Adding the TypeMeta here is not required anymore as the basereconciler_v2 ensures this is correctly set. Better remove it from here to make that fact clear.

```suggestion
```